### PR TITLE
[WIP] Messenger as command bus

### PR DESCRIPTION
**Pull request by Lutacon**
Initial integration to use Messenger as command bus and also to remove tactician and rabbit bundle dependency ;)

- [x] Create new command buses
- [x] Remove tactician dependency
- [ ] Create new transport
- [ ] Remove Old Sound RabbitMQ bundle dependency.

### CODE REVIEW

File: config/packages/messenger.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+framework:
+    messenger:
+        transports:
+            # Uncomment the following line to enable a transport named "amqp"
+             amqp: '%env(MESSENGER_TRANSPORT_DSN)%'
```
Comment (jorge07) [inline config/packages/messenger.yaml]: What if I don't want to send the queries to the transport?

File: config/services.yaml [hunk 2]
```diff
@@ -28,12 +28,12 @@ services:
         App\Application\Command\CommandHandlerInterface:
           public: true
           tags:
-              - { name: tactician.handler, typehints: true, bus: command }
+              - { name: messenger.message_handler, typehints: true, bus: messenger.bus.command }
```
Comment (jorge07) [inline config/services.yaml]: `typehints: true` should not be required

Comment (Lutacon) [reply config/services.yaml]: Mmmm Ok. 

Btw CommandHandlerInterface and QueryHandlerInterface, should not they have an invoke method?

File: config/packages/messenger.yaml (hunk 1 above)
Comment (Lutacon) [reply config/packages/messenger.yaml]: I'm going to look into that.

File: config/services.yaml (hunk 2 above)
Comment (Lutacon) [reply config/services.yaml]: Forget the question 🤣

File: .env.dist [hunk 3]
```diff
@@ -19,3 +19,7 @@ JWT_PUBLIC_KEY=%kernel.project_dir%/config/jwt/public.pem
 JWT_PASSPHRASE=development
 JWT_TTL=604800
 ###< lexik/jwt-authentication-bundle ###
+
+###> symfony/messenger ###
+MESSENGER_TRANSPORT_DSN=amqp://guest:guest@localhost:5672/%2f/messages
```
[the remark this answers is not in the document]
Comment (Lutacon) [reply .env.dist]: I need a package installed on PHP image, php7-amqp. I could make a PR on that image and add the package or make a new image. What do you prefer?

File: etc/deploy/Dockerfile [hunk 4]
```diff
@@ -2,6 +2,9 @@ FROM jorge07/alpine-php:7.2-dev-sf as builder
 
 WORKDIR /app
 
+RUN apk add -U --no-cache \
```
[the remark this answers is not in the document]
Comment (Lutacon) [reply etc/deploy/Dockerfile]: Good one

**Comment by jorge07** [comment]
Help wanted?

**Comment by Lutacon** [comment]
I was on vacation 😃  I hope to finish it this weekend. If I'm not able to finish last part I'm going to ask for your help

**Comment by karael** [comment]
Hi,
Takes care of messenger component it's not stable yet

**Comment by jorge07** [comment]
Any update on this @Lutacon? Any help wanted?

**Comment by Lutacon** [comment]
Let me see what I've got and maybe you can help finish it.

File: config/packages/messenger.yaml (hunk 1 above)
Comment (Lutacon) [reply config/packages/messenger.yaml]: By default, no message is sent to the transport. You need to specify which message goes to which transport.

File: config/packages/translation.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+framework:
+    default_locale: '%locale%'
+    translator:
+        paths:
+            - '%kernel.project_dir%/translations'
+        fallbacks:
+            - '%locale%'
```
Comment (jorge07) [inline config/packages/translation.yaml]: translations needed?

Comment (Lutacon) [reply config/packages/translation.yaml]: I have no idea how this end up here.

**Comment by db306** [comment]
Hi, I'm wondering if this is still something up to date given the discussion here [](https://github.com/jorge07/symfony-4-es-cqrs-boilerplate/issues/82) seems like this is not implemented anymore ?

**Comment by jorge07** [comment]
It is. This will help to replace tactician and rabbitmq bundle. I really want to see this working.

**Comment by karael** [comment]
So what's up ? I have some time to help

**Comment by jorge07** [comment]
@Lutacon I'll try myself this weekend this migration in a separated branch just to play. Any advice?

**Comment by Lutacon** [comment]
I have a problem with the serialization of the User, the Credentials VO is not being serialized. Tbh I didn't have much time to debug it.

Also, the messenger component changes a lot between 4.1 and 4.2

**Comment by waeyy** [comment]
Hi,

Any news concerning the implementation of Messenger as command bus ?

**Comment by db306** [comment]
I’m starting to work on it this week at work, it should technically be an interface implementation from the application layer on to infrastructure. I built it on Tactician at first should be able to translate it to messenger pretty easily.

**Comment by jorge07** [comment]
@Lutacon closing in favour of #132 ;)

**Pull request closed without merging**